packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -245,10 +144,15 @@ function getDocumentForPreloads(): ?Document {
   }
 }
 
-function getDocumentFromRoot(root: FloatRoot): Document {
+function getDocumentFromRoot(root: HoistableRoot): Document {
   return root.ownerDocument || root;
 }
 
+export function getHoistableRoot(container: Container): HoistableRoot {
+  // Flow thinks getRootNode returns Node but we know it is actualy either a Document or ShadowRoot
+  return ((container.getRootNode(): any): Document | ShadowRoot);
```

Comment:
@gnoff should we be using the `getRootNode` helper from above here? This is throwing a bunch of errors in internal classic component tests